### ISSUE TITLE
How to unwatch

### ISSUE THREAD

**Issue by IsaacLehman** [justin-schroeder/arrow-js#97]
Hey! 

Really loving ArrowJS - super easy to work with  :) 

One issue I have found is no way to "un watch". Take for example:
- You have some component with an onMount. When mounted you setup a `watch` on some reactive variable. 
- When the component get's disposed of, that `watch` keeps running.

It would be really helpful if the `watch` returned some sort of reference which we could then remove on unmount. 

A common time this happens is with multi-page apps. When spanning between pages, elements are constantly getting added and removed from the dom. After a bit you just have a bunch of old watch's sitting out there. 

Any suggestions are appreciated!
Thanks!

**Comment by justin-schroeder** [comment]
I agree, this seems like it would do the ticket. It is a significant breaking change too however — which will probably be required anyway whenever I get around to implementing something like this. There are a number of refactors that have been worked on that havent yet seen the light of day, but as soon as I get more time to tackle this unwatch will be a high priority!

**Comment by justin-schroeder** [comment]
Arrow > 1 now has an "onCleanup": https://arrow-js.com/api/#on-cleanup
